babel.config.json

```diff
@@ -0,0 +1,13 @@
+{
+  "presets": [],
+  "plugins": [
+    "@babel/plugin-syntax-dynamic-import"
+  ],
+  "env": {
+    "test": {
+      "plugins": [
+        "babel-plugin-dynamic-import-node"
+      ]
+    }
+  }
+}
```

Comment:
We already have a .babelrc.js: https://github.com/reduxjs/react-redux/blob/master/.babelrc.js

Reply:
Ah, I overlooked that. Let me move that over and push that up..

Reply:
moved the config to .babelrcjs and removed the babel.config.js, but it appears TravisCI is still not picking up that change. Tests run and pass locally..
![2021-01-29-132046_660x432_scrot](https://user-images.githubusercontent.com/2170844/106317995-bb972b80-6234-11eb-93dd-d19c5c87e80b.png)

Reply:
You should remove the babel changes. The version of Node we use (14.x) [already supports `import()` built-in](https://nodejs.org/dist/latest-v14.x/docs/api/esm.html#esm_import_expressions).

Reply:
And preset-env will also add the dynamic import syntax by default.

Reply:
This is needed for Jest. It is described in Jest documentation https://jestjs.io/docs/en/webpack.html#using-with-webpack-2
It is also discussed in this issue: https://github.com/facebook/jest/issues/2442

In that thread I just noticed this comment, which may be why TravisCI is failing but the tests run fine locally:
https://github.com/facebook/jest/issues/2442#issuecomment-471607499

> My tests were failing in CI but not locally the reason for this being that the NODE_ENV was set to production in the Dockerfile. The .babelrc env property is matched against the BABEL_ENV environment variable. Failing that it uses NODE_ENV. Jest will set NODE_ENV to "test" if it is null! Therefore because it was production it wouldn't set it to test and babel would not run the transform.